packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -1760,5 +1885,436 @@ describe("Controller helper integration test", function () {
+        liquidity: positionBefore.liquidity,
+        amount0Min: amount0Min,
+        amount1Min: amount1Min,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+  })
+      const wPowerPerpAmountInLp = (isWethToken0) ? amount1 : amount0;
+      const wethAmountInLp = (isWethToken0) ? amount0 : amount1;
+      //uniswap LPing often will give 1 wei less than expected, with the price of oSQTH need to do more than 1 wei due to rounding up the amount owed 
+      const squeethDesired = wPowerPerpAmountInLp.div(2)
+
+      const params = {
+        wPowerPerpPool: wSqueethPool.address,
+        tokenId: oldTokenId,
+        ethAmountToLp: BigNumber.from(0),
+        liquidity: oldPosition.liquidity,
+        wPowerPerpAmountDesired: squeethDesired,
+        wethAmountDesired: ethers.utils.parseUnits('0'),
+        amount0DesiredMin: BigNumber.from(0),
+        amount1DesiredMin: BigNumber.from(0),
+        limitPriceEthPerPowerPerp: BigNumber.from(0),
+        amount0Min: BigNumber.from(0),
+        amount1Min: BigNumber.from(0),
+        lowerTick: isWethToken0 ? -887220 : newTick,
+        upperTick: isWethToken0 ? newTick : 887220,
+        poolFee: 3000
+      }
+
+      await (positionManager as INonfungiblePositionManager).connect(depositor).approve(controllerHelper.address, oldTokenId);
+      await controllerHelper.connect(depositor).rebalanceWithoutVault(params);
+      tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+      const newTokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+      const newPosition = await (positionManager as INonfungiblePositionManager).positions(newTokenId);
+      const ownerOfUniNFT = await (positionManager as INonfungiblePositionManager).ownerOf(newTokenId); 
+
+      const [amount0New, amount1New] = await (positionManager as INonfungiblePositionManager).connect(depositor).callStatic.decreaseLiquidity({
+        tokenId: newTokenId,
+        liquidity: newPosition.liquidity,
+        amount0Min: BigNumber.from(0),
+        amount1Min: BigNumber.from(0),
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),
+      })
+
+      const wPowerPerpAmountInNewLp = (isWethToken0) ? amount1New : amount0New;
+      const wethAmountInNewLp = (isWethToken0) ? amount0New : amount1New;
+      // console.log('wPowerPerpAmountInLP',wPowerPerpAmountInLp.toString())
+      // console.log('wethAmountInLp', wethAmountInLp.toString())
+      // console.log('wPowerPerpAmountInNewLp',wPowerPerpAmountInNewLp.toString())
+      // console.log('wethAmountInNewLp', wethAmountInNewLp.toString())
+      expect(ownerOfUniNFT === depositor.address).to.be.true;
+      expect(wPowerPerpAmountInNewLp.sub(squeethDesired).lte(10)).to.be.true
+      expect(wethAmountInNewLp.eq(BigNumber.from(0))).to.be.true
+      expect(wPowerPerpAmountInNewLp.sub(wPowerPerpAmountInLp.div(2)).abs().lte(10)).to.be.true
```

Comment:
I think we should check the amount of ETH received as well from selling as well as withdrawn ETH from the LP